build.sc

```diff
@@ -47,4 +154,82 @@ object rocketchip extends common.CommonRocketChip {
   def hardfloatModule = hardfloatRocket
 
   def configModule = configRocket
+
+  object sanitytests extends Tests with TestModule.ScalaTest {
+    override def scalaVersion = T {
+      rocketchip.scalaVersion()
+    }
+
+    override def moduleDeps: Seq[JavaModule] = Seq(rocketchip)
+
+    override def scalacPluginClasspath = T { super.scalacPluginClasspath() }
+
+    def libraryResources = T.sources {
+      val x86Dir = T.ctx.dest
+      val riscv64Dir = T.ctx.dest / "riscv64"
+      val testDir: Path = os.pwd / "src" / "test" / "scala" / "sanitytests"
+      os.copy.into(testDir / "resources", x86Dir)
+      def resource(file: String): Path = x86Dir / RelPath(file)
+      os.proc("make", s"DESTDIR=${x86Dir}", "install").call(spike.compile())
+      os.proc("make", s"DESTDIR=${riscv64Dir}", "install").call(compilerrt.compile())
+      os.proc("make", s"DESTDIR=${riscv64Dir}", "install").call(musl.compile())
+      os.copy.into(pk.compile(), riscv64Dir)
+      // build hello world
+      val outputDirectory = x86Dir / "VerilatorTest"
+      os.remove.all(outputDirectory)
+      os.makeDir(outputDirectory)
+      os.proc(
+        "clang",
+        "-o", "hello",
+        s"${resource("resources/csrc/hello.c")}",
+        "--target=riscv64",
+        "-mno-relax",
+        "-nostdinc",
+        s"-I${resource("riscv64/usr/include")}",
+        "-fuse-ld=lld",
+        "-nostdlib",
+        s"${resource("riscv64/usr/lib/crt1.o")}",
+        s"${resource("riscv64/usr/lib/crti.o")}",
+        s"${resource("riscv64/usr/lib/riscv64/libclang_rt.builtins-riscv64.a")}",
+        s"${resource("riscv64/usr/lib/libc.a")}",
+        s"${resource("riscv64/usr/lib/crtn.o")}",
+        "-static",
+      ).call(outputDirectory)
+      // build bootrom
+      val linker = os.pwd / "bootrom" / "linker.ld"
+      val bootrom = os.pwd / "bootrom" / "bootrom.S"
+      val elf = outputDirectory / "bootrom.elf"
+      val bin = outputDirectory / "bootrom.bin"
+      val img = outputDirectory / "bootrom.img"
+      // format: off
+      os.proc(
+        "clang",
+        "--target=riscv64", "-march=rv64gc",
+        "-mno-relax",
+        "-static",
+        "-nostdlib",
+        "-Wl,--no-gc-sections",
+        "-fuse-ld=lld", s"-T$linker",
+        s"$bootrom",
+        "-o", elf
+      ).call()
+      os.proc(
+        "llvm-objcopy",
+        "-O", "binary",
+        elf,
+        bin
+      ).call()
+      os.proc(
+        "dd",
+        s"if=$bin",
+        s"of=$img",
+        "bs=128",
+        "count=1"
+      ).call()
+      T.ctx.dest
+    }
```

Comment:
make this as an external function, so that others can add programs as test payloads.